src/System.Management.Automation/utils/Encoding.cs

```diff
@@ -0,0 +1,399 @@
+    {
+
+        /// <summary>
+        /// Return the default PowerShell encoding which is UTF8 without a BOM.
+        /// There is no distinction between platforms
+        /// </summary>
+        public static Encoding GetDefaultEncoding()
+        {
+            return utf8NoBom;
+        }
+
+        /// <summary>
+        /// translate a FileEncoding to an actual System.Text.Encoding
+        /// <param name="TextEncoding">The enum value</param>
+        /// <returns>System.Text.Encoding</returns>
+        /// </summary>
+        public static Encoding GetEncoding(FileEncoding TextEncoding)
+        {
+            System.Text.Encoding result = GetDefaultEncoding();
+            switch ( TextEncoding )
+            {
+                case FileEncoding.String:
+                    result = Encoding.Unicode;
+                    break;
+
+                case FileEncoding.Unicode:
+                    result = Encoding.Unicode;
+                    break;
+
+                case FileEncoding.BigEndianUnicode:
+                    result = Encoding.BigEndianUnicode;
+                    break;
+
+                case FileEncoding.UTF8BOM:
+                    result = Encoding.UTF8; // The default UTF8 encoder includes the BOM
+                    break;
+
+                case FileEncoding.Byte:
+                    result = Encoding.Unicode;
+                    break;
+
+                case FileEncoding.UTF8:
+                case FileEncoding.UTF8NoBOM:
+                    result = utf8NoBom;
+                    break;
+
+                case FileEncoding.UTF7:
+                    result = Encoding.UTF7;
+                    break;
+
+                case FileEncoding.UTF32:
+                    result = Encoding.UTF32;
+                    break;
+
+                case FileEncoding.BigEndianUTF32:
+                    // This can possibly throw, but if so, we can't provide
+                    // the encoding which the user requested, so we should fail
+                    result = Encoding.GetEncoding("utf-32BE");
+                    break;
+
+                case FileEncoding.Ascii:
+                    result = Encoding.ASCII;
+                    break;
+
+                case FileEncoding.Default:
+                    result = GetDefaultEncoding();
+                    break;
+
+                case FileEncoding.Oem:
+                    result = ClrFacade.GetOEMEncoding();
+                    break;
+
+                default:
+                    break;
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Retrieve the encoding based on the Cmdlet and the Encoding
+        /// <param name="cmdlet">The cmdlet of interest</param>
+        /// <param name="encoding">The Encoding parameter value</param>
+        /// <returns>System.Text.Encoding</returns>
+        /// </summary>
+        public static Encoding GetEncoding(Cmdlet cmdlet, FileEncoding encoding)
```

Comment:
i fixed the public api parameter validation but I disagree about the throw - if I don't have access to session state, I can't retrieve the value I can still return that no encoding preference has been set